### PR TITLE
Pull Request for Issue1967: Resolving differences in utilities controls on BioXAS

### PR DESCRIPTION
**Pull request by helfrij**
- Modified BioXASShutters. Now follows same model as BioXASValves--it is now a collection of shutters subcontrols.
- Removed old BioXASShuttersGroup class.
- BioXASFrontEndShutters now inherits from BioXASShutters.
- Created new general utilities class, BioXASUtilities, to replace BioXASBeamlineUtilities and the beamline-specific subclasses.
- Added new methods to BioXASBeamline to add, remove, or clear utilities components: shutters, valves, beampath valves, ion pumps, flow switches, flow transducers, temperature monitors, pressure monitors.
- Added instances of BioXASValves and BioXASShutters to BioXASBeamStatus. The beam status now has methods for adding/removing/clearing valves and shutters subcontrols.
- Created ultra-basic utilities view, added it to the app controller.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you take a look at this pull request?

File: source/beamline/BioXAS/BioXASBeamStatus.cpp [hunk 1]
```diff
@@ -18,6 +40,18 @@ BioXASBeamStatus::~BioXASBeamStatus()
 
 }
 
+bool BioXASBeamStatus::isConnected() const
+{
+	bool connected = (
+				shutters_ && shutters_->isConnected() &&
+				valves_ && valves_->isConnected() &&
+				mirrorMaskState_ && mirrorMaskState_->isConnected() &&
+				monoMaskState_ && monoMaskState_->isConnected()
```
Comment (XuetaoLiu-CLS) [inline source/beamline/BioXAS/BioXASBeamStatus.cpp]: @helfrij I want to confirm the logic here. If we forgot to initialize the `mirrorMaskState_` and/or `monoMaskState_`. `isConnected()` will always return false. So far, both the main/side bioxas are using BioXASBeamStatus and initialized the mask control. Will imaging use `BioXASBeamStatus` ?

Comment (helfrij) [inline source/beamline/BioXAS/BioXASBeamStatus.cpp]: Not as written. The Imaging beamline doesn't have an M1 mask or a mono mask--only Side and Main have those features.

Comment (XuetaoLiu-CLS) [inline source/beamline/BioXAS/BioXASBeamStatus.cpp]: Thanks for the clarification.

**Comment by XuetaoLiu-CLS** [comment]
Looks good. A good way to organize the shutters/valves of the beamline. The codes are much clearer.

Time Estimate: 30 minutes

**Comment by XuetaoLiu-CLS** [comment]
@helfrij you need to resolve the merge conflicts.

**Comment by helfrij** [comment]
The merge conflicts should be resolved. Thanks @XuetaoLiu-CLS!

File: source/beamline/BioXAS/BioXASUtilitiesGroup.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	addOption(Bad, "Bad", true);
+	addOption(Good, "Good", true);
+}
+
+BioXASUtilitiesGroup::~BioXASUtilitiesGroup()
+{
+
+}
+
+bool BioXASUtilitiesGroup::isBad() const
```
Comment (dretrex) [inline source/beamline/BioXAS/BioXASUtilitiesGroup.cpp]: What about `isHappy()`? ;)

**Comment by dretrex** [comment]
Looks good to me.  Really slick.

Time Estimate: 10 minutes

File: source/beamline/BioXAS/BioXASUtilitiesGroup.cpp (hunk 2 above)
Comment (helfrij) [inline source/beamline/BioXAS/BioXASUtilitiesGroup.cpp]: Happiness is not an option for utilities on BioXAS. It's a black and white world over here!

**Pull request merged**